securesystemslib/interface.py

```diff
@@ -120,237 +104,194 @@ def get_password(prompt='Password: ', confirm=False):
-  if password is not None:
-    securesystemslib.formats.PASSWORD_SCHEMA.check_match(password)
+  Raises:
+    UnsupportedLibraryError: pyca/cryptography is not available.
+    FormatError: Arguments are malformed.
+    ValueError: Both a 'password' is passed and 'prompt' is true.
+    StorageError: Key file cannot be read.
+    CryptoError: Key cannot be parsed.
 
-    # TODO: PASSWORD_SCHEMA should be securesystemslib.schema.AnyString(min=1)
-    if not len(password):
-      raise ValueError('Password must be 1 or more characters')
-
-  elif prompt:
-    # Password confirmation disabled here, which should ideally happen only
-    # when creating encrypted key files (i.e., improve usability).
-    # It is safe to specify the full path of 'filepath' in the prompt and not
-    # worry about leaking sensitive information about the key's location.
-    # However, care should be taken when including the full path in exceptions
-    # and log files.
-    # NOTE: A user who gets prompted for a password, can only signal that the
-    # key is not encrypted by entering no password in the prompt, as opposed
-    # to a programmer who can call the function with or without a 'password'.
-    # Hence, we treat an empty password here, as if no 'password' was passed.
-    password = get_password('Enter a password for an encrypted RSA'
-        ' file \'' + TERM_RED + filepath + TERM_RESET + '\': ',
-        confirm=False) or None
+  Returns:
+    An RSA private key object conformant with 'RSAKEY_SCHEMA'.
 
-  if password is not None:
-    # This check will not fail, because a mal-formatted passed password fails
-    # above and an entered password will always be a string (see get_password)
-    # However, we include it in case PASSWORD_SCHEMA or get_password changes.
-    securesystemslib.formats.PASSWORD_SCHEMA.check_match(password)
+  """
+  securesystemslib.formats.PATH_SCHEMA.check_match(filepath)
+  securesystemslib.formats.RSA_SCHEME_SCHEMA.check_match(scheme)
 
-  else:
-    logger.debug('No password was given. Attempting to import an'
-        ' unencrypted file.')
+  password = _get_key_file_decryption_password(password, prompt, filepath)
 
   if storage_backend is None:
     storage_backend = securesystemslib.storage.FilesystemBackend()
 
-  # Read the contents of 'filepath' that should be a PEM formatted private key.
   with storage_backend.get(filepath) as file_object:
     pem_key = file_object.read().decode('utf-8')
 
-  # Convert 'pem_key' to 'securesystemslib.formats.RSAKEY_SCHEMA' format.
-  # Raise 'securesystemslib.exceptions.CryptoError' if 'pem_key' is invalid.
-  # If 'password' is None decryption will be omitted.
-  rsa_key = securesystemslib.keys.import_rsakey_from_private_pem(pem_key,
-      scheme, password)
+  # Optionally decrypt and convert PEM-encoded key to 'RSAKEY_SCHEMA' format
+  rsa_key = securesystemslib.keys.import_rsakey_from_private_pem(
+      pem_key, scheme, password)
```

Comment:
I have a small gripe with the way we set up these two functions. It feels to me that the pattern I most of the time do is:

```
generate a key
use the key right away
```

It is rather confusing to me that the filepath is returned, rather than the generated key itself, specially given that there is no default filepath (so we don't have to provide users any disambiguation). So with this in mind, what's the added value of returning the filename rather than the key itself?

Reply:
Agreed. This was just not a focus of this PR.

Reply:
Also, it is already possible to do that one level deeper in the abstraction hierarchy (see `securesystemslib.keys.generate_*_key`).

Reply:
Fair, I still wonder if we'd like to change the return value for the key instead of the filepath?